### PR TITLE
Update roadmap for 2024 and beyond

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:
Updating the roadmap based on the recent refinement sessions.

cc @kubernetes/release-engineering @kubernetes/sig-release-leads 
#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None

### CODE REVIEW

File: roadmap.md [hunk 1]
```diff
@@ -50,51 +43,26 @@ state.
...
-   Outcome: Being able to promote files as artifacts and using this mechanism
-   for Kubernetes releases.
-
-1. **Define and collect metrics about Kubernetes releases (Introspectable)**
+1. **Make Kubernetes releases more flexible**
 
-   https://github.com/kubernetes/sig-release/issues/1527
+   Outcome: Ensure we migrate away from gs://kubernetes-release, use a
+   community-owned Object Storage bucket for binaries releases, serve the
+   binaries published through a CDN service.
```
Comment (saschagrunert) [inline roadmap.md:50]: @ameukam is it fine if we make this a bit less specific? This way we could integrate topics like:

- Allow creating kubernetes releases from (private) forks
- Add parameters to krel to support other destinations (buckets)

Comment (ameukam) [reply roadmap.md:50]: That will increase the timeframe to implement this. 
We may end up in a situation where the `gs://kubernetes-release` bucket is unavailable for the release process.
Better to do this ASAP (in 1.30).

Comment (saschagrunert) [reply roadmap.md]: Good, reverted back to the original state.

File: roadmap.md [hunk 2]
```diff
@@ -31,15 +25,14 @@ state.
...
+   Outcome: Provide a fast and error-resistant workflow for artifact promotion
+   compliant with SLSA.
 
-   Enhancement: https://github.com/kubernetes/enhancements/issues/3027
+   Enhancement: was https://github.com/kubernetes/enhancements/issues/3027
 
-   Project board: https://github.com/orgs/kubernetes/projects/138
+   Project board: https://github.com/orgs/kubernetes/projects/171
 
 1. **Moving deb/rpm package builds to community infrastructure (Consumable)**
```
Comment (xmudrii) [inline roadmap.md:37]: This is in fact done, so I think we should rephrase this goal. I would split it into two deliverables:

- Making deb and rpm package builds more robust
  - The remaining work that we have is very well described with the provided robustness definition, e.g. writing tests, enabling reproducible builds...
  - Outcome can be similar to the outcome of artifact validation, e.g. provide a fast and error-resistant workflow for building and publishing packages to the community-owned infrastructure
- Enable other Kubernetes subprojects to use our packages infrastructure
  - This is about flexibility and it should cover efforts like building a CLI, library, writing documentation, etc. We don't have a KEP for this, but we can reuse the existing project board for now

Comment (saschagrunert) [reply roadmap.md:37]: @xmudrii considering current resources, should we move both into the `Todo` section?

Comment (saschagrunert) [reply roadmap.md]: Updated the roadmap to reflect the current state.

**Comment by saschagrunert** [comment]
@kubernetes/release-managers @kubernetes/sig-release PTAL again

**Pull request merged**